homeassistant/components/konnected/__init__.py

```diff
@@ -430,20 +482,45 @@ def binary_value(state, activation):
         if device is None:
             return self.json_message('unregistered device',
                                      status_code=HTTP_BAD_REQUEST)
-        pin_data = device[CONF_BINARY_SENSORS].get(pin_num)
+        pin_data = device[CONF_BINARY_SENSORS].get(pin_num) or \
+            next((s for s in device[CONF_SENSORS] if s[CONF_PIN] == pin_num),
+                 None)
 
         if pin_data is None:
             return self.json_message('unregistered sensor/actuator',
                                      status_code=HTTP_BAD_REQUEST)
 
-        entity_id = pin_data.get(ATTR_ENTITY_ID)
-        if entity_id is None:
-            return self.json_message('uninitialized sensor/actuator',
-                                     status_code=HTTP_NOT_FOUND)
-        state = bool(int(state))
-        if pin_data.get(CONF_INVERSE):
-            state = not state
+        if state:
```

Comment:
This part of the code would benefit of using our handler registry util.

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/util/decorator.py

We would register one handler per `'state'`, `'temp'`, `'humi'` and `'addr'`. Then iterate over registered handlers, passing the `payload` and the `pin_data` when calling each handler.

This will clean up this part significantly and make for easier extension.